### DISCUSSION TITLE
why cant use "

### DISCUSSION THREAD

**Discussion by YongMingMao** [microsoft/WSL#8200] [General]
### Version

Microsoft Windows [版本 10.0.22000.556]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

 5.10.16

### Distro Version

Ubuntu

### Repro Steps

i want to install unzip
put into command "apt install unzip" 
it shows 
"
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package unzip
"
and 
I want use command " apt-get update"
it return
"Err:1 https://mirrors.ustc.edu.cn/ubuntu focal InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Could not connect to mirrors.ustc.edu.cn:443 (202.141.176.110), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out
Err:2 https://mirrors.ustc.edu.cn/ubuntu focal-updates InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
Err:3 https://mirrors.ustc.edu.cn/ubuntu focal-backports InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
Err:4 https://mirrors.ustc.edu.cn/ubuntu focal-security InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
Reading package lists... Done
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Could not connect to mirrors.ustc.edu.cn:443 (202.141.176.110), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal-updates/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal-backports/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal-security/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
W: Some index files failed to download. They have been ignored, or old ones used instead."

It seems I can not install any app in ubuntu

### Expected Behavior

i want to  install app in this ubuntu

### Actual Behavior

it return
"Err:1 https://mirrors.ustc.edu.cn/ubuntu focal InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Could not connect to mirrors.ustc.edu.cn:443 (202.141.176.110), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out
Err:2 https://mirrors.ustc.edu.cn/ubuntu focal-updates InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
Err:3 https://mirrors.ustc.edu.cn/ubuntu focal-backports InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
Err:4 https://mirrors.ustc.edu.cn/ubuntu focal-security InRelease
  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
Reading package lists... Done
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Could not connect to mirrors.ustc.edu.cn:443 (202.141.176.110), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.95.94), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (202.38.93.95), connection timed out Could not connect to mirrors.ustc.edu.cn:443 (116.62.60.40), connection timed out
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal-updates/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal-backports/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
W: Failed to fetch https://mirrors.ustc.edu.cn/ubuntu/dists/focal-security/InRelease  Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::110). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:95::94). - connect (101: Network is unreachable) Cannot initiate the connection to mirrors.ustc.edu.cn:443 (2001:da8:d800:931::95). - connect (101: Network is unreachable)
W: Some index files failed to download. They have been ignored, or old ones used instead."

**Comment by yongtong19** [comment]
It seems that your network condition is poor. Have you ever tried to ping mirrors.ustc.edu.cn or other domain names in China like baidu.com?

**Reply by theassyrian** [reply]
The network vanished outside local VM. It happened to me also and I am still trying to determine the cause.
Whatever virtual switch microsoft uses for external vm connections is not working anymore
